site/app/libraries/SessionManager.php

```diff
@@ -40,17 +46,27 @@ public function __construct(Core $core) {
      * @return bool|string
      */
     public function getSession(string $session_id) {
-        $this->session = $this->core->getQueries()->getSession($session_id);
+        $em = $this->core->getSubmittyEntityManager();
+        /** @var SessionRepository $repo */
+        $repo =  $em->getRepository(Session::class);
+        $expressionBuilder = Criteria::expr();
+        $this->session = $repo->matching(new Criteria(
+            $expressionBuilder->andX(
+                $expressionBuilder->eq("session_id", $session_id),
+                $expressionBuilder->gt("session_expires", $this->core->getDateTimeNow())
+            )
+        ))->first();
```

Comment:
Could we instead put this logic inside the repository? Then we can just have a function like `getActiveSessionById` or something.